egs/zeroth_korean/s5/local/nnet3/run_ivector_common.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# this script contains some common (shared) parts of the run_nnet*.sh scripts.
+. cmd.sh
+
+
+stage=0
+gmmdir=exp/tri4
+speed_perturb=false
+trainset=train_clean
+
+set -e
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if [ "$speed_perturb" == "true" ]; then
+  if [ $stage -le 1 ]; then
+    echo "$0: preparing directory for speed-perturbed data"
+    #Although the nnet will be trained by high resolution data, we still have to perturbe the normal data to get the alignment
+    # _sp stands for speed-perturbed
+    for datadir in ${trainset} ; do
+	  utils/data/perturb_data_dir_speed_3way.sh data/${datadir} data/${datadir}_sp 
+
+      mfccdir=mfcc_perturbed
+      steps/make_mfcc.sh --cmd "$train_cmd" --nj 40 \
+        data/${datadir}_sp exp/make_mfcc/${datadir}_sp $mfccdir || exit 1;
+      steps/compute_cmvn_stats.sh data/${datadir}_sp exp/make_mfcc/${datadir}_sp $mfccdir || exit 1;
+      utils/fix_data_dir.sh data/${datadir}_sp
+    done
+  fi
+
+  if [ $stage -le 2 ]; then
+	echo "$0: aligning with the perturbed low-resolution data"
+    #obtain the alignment of the perturbed data
+    steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" \
+      data/${trainset}_sp data/lang_nosp ${gmmdir} ${gmmdir}_ali_${trainset}_sp || exit 1
+  fi
+  trainset=${trainset}_sp
+fi
+
+if [ $stage -le 3 ]; then
+  # Create high-resolution MFCC features (with 40 cepstra instead of 13).
+  # this shows how you can split across multiple file-systems.  we'll split the
+  # MFCC dir across multiple locations.  You might want to be careful here, if you
+  # have multiple copies of Kaldi checked out and run the same recipe, not to let
+  # them overwrite each other.
+
+  echo "$0: creating high-resolution MFCC features"  
+  for datadir in ${trainset} ; do
+    utils/copy_data_dir.sh data/$datadir data/${datadir}_hires
+    steps/make_mfcc.sh --nj 40 --mfcc-config conf/mfcc_hires.conf \
+      --cmd "$train_cmd" data/${datadir}_hires  || exit 1;
+    steps/compute_cmvn_stats.sh data/${datadir}_hires  || exit 1;
+  done
+
+  # We need to build a small system just because we need the LDA+MLLT transform
```

Comment:
these comments are out of date, we are using PCA not LDA+MLLT.

Reply:
yes. I corrected it.